sourcegraph/cody
# Autocomplete: format completions on accept

## Context

- Formats autocomplete suggestions using the available formatter on accept.
- Addresses [user feedback](https://sourcegraph.slack.com/archives/C05AGQYD528/p1701771083790929): "_A dev told us that Cody was suggesting tabs for python code whereas, he was expecting spaces_"
- Closes https://github.com/sourcegraph/cody/issues/2252

## Test plan

1. Write code using tabs instead of spaces.
2. Ensure that you have a configured formatter that prefers spaces.
3. Generate a multiline completion. StarCoder will (most probably) match the indentation style of the existing code (tabs).
4. Accept the completion and ensure that tabs are converted to spaces automatically. 
5. Ensure that only the inserted completion is formatted.
